src/R/Core/Impl/Tokens/Number.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Globalization;
+
+namespace Microsoft.R.Core.Tokens {
+    public static class Number {
+        private const NumberStyles _hexStyle = NumberStyles.HexNumber | NumberStyles.AllowHexSpecifier;
+        private const NumberStyles _integerStyle = NumberStyles.Integer;
+        private const NumberStyles _doubleStyle = NumberStyles.Number | NumberStyles.AllowExponent;
+
+        public static bool TryParse(string text, out double doubleResult, bool allowLSuffix = true) {
+            doubleResult = Double.NaN;
+            int sign = 1;
+
+            if (text.Length == 0) {
+                return false;
+            }
+            if (allowLSuffix && text[text.Length - 1] == 'L') {
+                text = text.Substring(0, text.Length - 1);
+            }
+
+            if (text.Length == 0) {
+                return false;
+            }
+            if (text[0] == '-' || text[0] == '+') {
```

Comment:
So, I am not changing prefix handling to operator just yet, but will file a separate task